src/processor/map/map_insert.cpp

```diff
@@ -31,26 +31,20 @@ static std::vector<DataPos> populateReturnVectorsPos(const LogicalInsertInfo& in
 std::unique_ptr<NodeInsertExecutor> PlanMapper::getNodeInsertExecutor(const LogicalInsertInfo* info,
     const Schema& inSchema, const Schema& outSchema) const {
     auto storageManager = clientContext->getStorageManager();
-    auto node = ku_dynamic_cast<Expression*, NodeExpression*>(info->pattern.get());
-    auto nodeTableID = node->getSingleTableID();
-    auto table = ku_dynamic_cast<Table*, NodeTable*>(storageManager->getTable(nodeTableID));
+    auto catalog = clientContext->getCatalog();
+    auto transaction = clientContext->getTx();
+    auto& node = info->pattern->constCast<NodeExpression>();
+    auto nodeTableID = node.getSingleTableID();
+    auto table = storageManager->getTable(nodeTableID)->ptrCast<NodeTable>();
     std::unordered_set<RelTable*> fwdRelTablesToInit;
     std::unordered_set<RelTable*> bwdRelTablesToInit;
-    auto tableCatalogEntry =
-        clientContext->getCatalog()->getTableCatalogEntry(clientContext->getTx(), nodeTableID);
-    auto nodeTableSchema =
-        ku_dynamic_cast<TableCatalogEntry*, NodeTableCatalogEntry*>(tableCatalogEntry);
-    auto fwdRelTableIDs = nodeTableSchema->getFwdRelTableIDSet();
-    auto bwdRelTableIDs = nodeTableSchema->getBwdRelTableIDSet();
-    for (auto relTableID : fwdRelTableIDs) {
-        fwdRelTablesToInit.insert(
-            ku_dynamic_cast<Table*, RelTable*>(storageManager->getTable(relTableID)));
+    for (auto relTableID : catalog->getFwdRelTableIDs(transaction, nodeTableID)) {
```

Comment:
I don't think we need `fwdRelTablesToInit` and `bwdRelTablesToInit` in Insert operation anymore. Can u double check and clean it up?